src/uct/tcp/sockcm/sockcm_md.c

```diff
@@ -44,29 +44,112 @@ ucs_status_t uct_sockcm_md_query(uct_md_h md, uct_md_attr_t *md_attr)
     return UCS_OK;
 }
 
+static int uct_sockcm_is_addr_route_resolved(int sock_id, struct sockaddr *addr,
+                                            int addrlen)
+{
+    char host[UCS_SOCKADDR_STRING_LEN];
+    char serv[UCS_SOCKADDR_STRING_LEN];
+    int ret = -1;
+
+    ret = getnameinfo(addr, addrlen, host, UCS_SOCKADDR_STRING_LEN, serv,
+                      UCS_SOCKADDR_STRING_LEN, NI_NAMEREQD);
+    if (0 != ret) {
+        ucs_debug("getnameinfo error : %s\n", gai_strerror(ret));
+        return 0;
+    }
+
+    return 1;
+}
+
 int uct_sockcm_is_sockaddr_accessible(uct_md_h md, const ucs_sock_addr_t *sockaddr,
                                       uct_sockaddr_accessibility_t mode)
 {
-    return 0;
+    uct_sockcm_md_t *sockcm_md = ucs_derived_of(md, uct_sockcm_md_t);
+    int is_accessible = 0;
+    int sock_id = -1;
+    char ip_port_str[UCS_SOCKADDR_STRING_LEN];
+    struct sockaddr *param_sockaddr = NULL;
+    struct sockaddr_in *addr;
+    socklen_t sockaddr_len;
+
+    param_sockaddr = (struct sockaddr *) sockaddr->addr;
+
+    if ((mode != UCT_SOCKADDR_ACC_LOCAL) && (mode != UCT_SOCKADDR_ACC_REMOTE)) {
+        ucs_error("Unknown sockaddr accessibility mode %d", mode);
+        return 0;
+    }
+
+    sock_id = socket(param_sockaddr->sa_family, SOCK_STREAM, 0);
+    if (-1 == sock_id) {
+        return 0;
+    }
+
+    if (mode == UCT_SOCKADDR_ACC_LOCAL) {
+        addr = (struct sockaddr_in *) param_sockaddr;
+        if (addr->sin_family == AF_INET) {
+            ucs_debug("family = AF_INET");
+            sockaddr_len = sizeof(struct sockaddr_in);
+        } else if (addr->sin_family == AF_INET6) {
+            ucs_debug("family = AF_INET6");
+            sockaddr_len = sizeof(struct sockaddr_in6);
+        } else {
+            ucs_debug("family != AF_INET and != AF_INET6");
+            sockaddr_len = -1;
+        }
+        ucs_debug("addr_len = %ld", (long int) sockaddr_len);
+
+        if (bind(sock_id, param_sockaddr, sockaddr_len)) {
+            ucs_debug("bind(addr = %s) failed: %m",
+                      ucs_sockaddr_str((struct sockaddr *)sockaddr->addr,
+                                       ip_port_str, UCS_SOCKADDR_STRING_LEN));
+            goto out_destroy_id;
+        }
+
+        if (ucs_sockaddr_is_inaddr_any(param_sockaddr)) {
+            is_accessible = 1;
+            goto out_print;
+        }
+    }
+
+    is_accessible = uct_sockcm_is_addr_route_resolved(sock_id, param_sockaddr,
```

Comment:
is the concept of 'route_resolved' relevant for sockcm? I think it only applies to rdmacm since it has the rdma_route_resolve function call that is followed by the RDMA_CM_EVENT_ROUTE_RESOLVED

Reply:
I don't think it's needed. I'm happy to drop this.